### PR TITLE
CB2 2829 filter out cancelled test results

### PR DESCRIPTION
**Pull request by Ian-Hodges**
ATFs were still getting charged despite the test being cancelled. Discarding cancelled results will resolve this issue.

### CODE REVIEW

File: src/utils/dataFormatter.ts [hunk 1]
```diff
@@ -7,22 +7,24 @@ import { TestActivity } from './testActivity';
...
-    testResult: testType.testResult,
-    certificateNumber: testType.certificateNumber,
-    testTypeName: testType.name,
-    vehicleType: data.vehicleType,
-    testerName: data.testerName,
-    testerStaffId: data.testerStaffId,
-    testResultId: data.testResultId,
-  }));
+  const testActivities: TestActivity[] = data.testTypes
+    .filter(() => data.testStatus !== 'cancelled')
```
Comment (NiallBunting) [inline src/utils/dataFormatter.ts:11]: Should we be using enums for values of the testStatus?

Also this seems like logic that's not exactly around formating data, what would happen if you wanted to reuse this formatter in future to get formatted cancelled data?

**Review by NiallBunting — COMMENTED** [review]
I'm a bit out of the loop with some of this stuff, anyway...

File: src/utils/dataFormatter.ts (hunk 1 above)
Comment (Ian-Hodges) [reply src/utils/dataFormatter.ts:11]: The testStatus is a string and comes in from the DynamoDB stream. We don't have control over that.

I believe the formater will only have one use case. We could easily refactor later if we did want to start getting different events based on testStatus or some other property.

Comment (Ian-Hodges) [reply src/utils/dataFormatter.ts]: The more I thought about it, the more it bothered me :-)
Renamed a few things to better reflect their usage.

**Review by premghinde — APPROVED** [review]

**Review by andrewf-bjss — APPROVED** [review]

**Review by danielchildsbjss — APPROVED** [review]

**Pull request merged**